corehq/apps/sms/models.py

```diff
@@ -775,6 +777,67 @@ class Meta:
         app_label = "sms"
 
 
+class CommConnectCaseSQL(CommCareCaseSQL, CommCareMobileContactMixin):
+    class Meta:
+        app_label = "sms"
+        proxy = True
+
+    def get_phone_info(self):
+        PhoneInfo = namedtuple(
+            'PhoneInfo',
+            [
+                'requires_entry',
+                'phone_number',
+                'sms_backend_id',
+                'ivr_backend_id',
+            ]
+        )
+        contact_phone_number = self.get_case_property('contact_phone_number')
+        contact_phone_number = apply_leniency(contact_phone_number)
+        contact_phone_number_is_verified = self.get_case_property('contact_phone_number_is_verified')
+        contact_backend_id = self.get_case_property('contact_backend_id')
+        contact_ivr_backend_id = self.get_case_property('contact_ivr_backend_id')
+
+        requires_entry = (
+            contact_phone_number and
+            contact_phone_number != '0' and
+            not self.closed and
+            not self.is_deleted and
+            # For legacy reasons, any truthy value here suffices
+            contact_phone_number_is_verified
+        )
+
+        return PhoneInfo(
+            requires_entry,
+            contact_phone_number,
+            contact_backend_id,
+            contact_ivr_backend_id
+        )
+
+    def get_time_zone(self):
+        return self.get_case_property('time_zone')
+
+    def get_language_code(self):
+        return self.get_case_property('language_code')
+
+    def get_email(self):
+        return self.get_case_property('commcare_email_address')
+
+    @property
+    def raw_username(self):
+        return self.case_name
+
+    @classmethod
+    def wrap_as_commconnect_case(cls, case):
+        """
+        Takes a CommCareCaseSQL and returns it as a CommConnectCaseSQL.
+        """
+        # Having an extra lookup to do this isn't great.
+        # TODO: Try to replace all references to wrap_as_commconnect_case to
+        # avoid the extra lookup.
+        return CaseAccessorSQL.get_case(case.case_id, cls=cls)
```

Comment:
i see. i think we previously decided against making CommConnectCase a mixin on CommCareCase because casexml was a submodule and we didn't want it to have dependencies on corehq code.  But now that casexml is merged in, I think making it a mixin would actually be a cleaner way to do this (and is the same way we do it for users that are messaging contacts)

@snopoke @czue do you see any issue with me making CommCareCaseSQL and CommCareCase direct implementers of the CommCareMobileContactMixin the same way we do CommCareUser, or would you prefer i used a wrapper class? I'm actually thinking of keeping the CommConnectCase class and making it inherit from CommCareMobileContactMixin and implement all the methods, and then just making CommConnectCase be a mixin on CommCareCaseSQL, that way all the messaging contact code stays separate.